compiler/rustc_hir_analysis/src/check/intrinsicck.rs

```diff
@@ -299,6 +299,28 @@ impl<'a, 'tcx> InlineAsmCtxt<'a, 'tcx> {
             }
         }
 
+        match *ty.kind() {
+            ty::RawPtr(_, hir::Mutability::Mut) if asm.options.contains(InlineAsmOptions::READONLY) =>
+                self
+                    .tcx
+                    .dcx()
+                    .struct_span_warn(expr.span, "passing a mutable pointer to asm! block with 'readonly' option.")
+                    .with_note("`readonly` means that no memory write happens inside the asm! block.")
+                    .with_note("This is not limited to global variables, it also includes passed pointers.")
+                    .with_note("If passing this mutable pointer is intentional, remove the `readonly` attribute.")
```

Comment:
I don't think this message is appropriate: if passing a mutable pointer is intentional, we should only suggest something which suppresses the warning, not something that changes the semantics of the `asm!`.

The problem here is that I don't see an obvious way to suppress the warning. Perhaps this lint would be better in clippy instead rather than something that generally applies to all Rust code?

Reply:
Hmm, now that I'm thinking about it, maybe you're right - `*mut` pointer should anyway implicitly cast to `*const` ones. What do you think about the other case with `*const/mut` and `nomem`?